src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+                    f"Knowledge base. This might reduce the efficiency of the algo."
+                )
+            )
+        # TODO: Should we have this dimension be larger than the current number of
+        # experiments in the KB, in case some get added in the future?
+        max_task_id = self.knowledge_base.n_stored_experiments + 1
+        task_label_dimension = Categorical(
+            "task_id", list(range(0, max_task_id)), default_value=0,
+        )
+        space_without_task_ids = Space(space.copy())
+        space_with_task_ids = Space(space.copy())
+        space_with_task_ids.register(task_label_dimension)
+        assert "task_id" in space_with_task_ids
+        assert "task_id" not in space_without_task_ids
+
+        self.algorithm = PrimaryAlgo(
+            space=space_with_task_ids, algorithm_config=algorithm_config
+        )
+        self._space = space_without_task_ids
+        self.current_task_id = 0
+        assert "task_id" not in self._space
+        assert "task_id" not in self.space
+        # TODO: For some optimizers this exact key won't be found, there is a View(OneHotEncode(...))
+        # assert "task_id" in self.algorithm.transformed_space, self.algorithm.transformed_space
+        # assert "task_id" in self.algorithm.space
+        self.transformed_space = self.space
+
+    def suggest(self, num: int = 1) -> List[Point]:
+        """Suggest a `num` of new sets of parameters.
+
+        :param num: how many sets to be suggested.
+
+        .. note:: New parameters must be compliant with the problem's domain
+           `orion.algo.space.Space`.
+        """
+        points = self.algorithm.suggest(num=num)
+        if not self._enabled:
+            return points
+
+        # Remove the task ids from these points:
+        # NOTE: These predicted task ids aren't currently used.
+
+        points_without_task_ids = [self._remove_task_id(point) for point in points]
+        for point in points_without_task_ids:
+            if point not in self.space:
+                raise ValueError(
+                    textwrap.dedent(
+                        f"""\
+                        Point is not contained in space:
+                        Point: {point}
+                        Space: {self.space}
+                        """
+                    )
+                )
+        return points_without_task_ids
+
+    def observe(
+        self, points: List[Point], results: Union[List[float], List[Dict]]
+    ) -> None:
+        """Observe evaluation `results` corresponding to list of `points` in
+        space.
+
+        .. seealso:: `orion.algo.base.BaseAlgorithm.observe`
+        """
+        # Add the task ids to the points, because we assume that the points passed to
+        # `observe` are from the current task.
+        if self._enabled:
+            # Add the task ids to the points.
+            points = [
+                self._add_task_id(point, self.current_task_id) for point in points
```

Comment:
This method _add_task_id is not implemented by default? Should this method also ignore the passed task_id if the point already contains it? I assume observe will be called with points containing the task_id during warm-starting.